### DISCUSSION TITLE
Is there a guide on getting TripleA into IDEA?

### DISCUSSION THREAD

**Discussion by zlefin** [triplea-game/triplea#9251] [New Contributor]
hold on, found something else to try and seeing if it works. nm it didn't work

I'm switching to IDEA from Eclipse (since I'd never learned eclipse that well anyways).

I'm having trouble getting all the configurations right to be able to actually test things on my own system.

In particular I get the error:
Gradle sync failed: Cause: com/install4j/gradle/Install4jPlugin has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 53.0 (9 s 725 ms)

and I'm not sure why.  I downloaded jdk 16, and adjusted the module settings so that it should use jdk 16 for everything, but it still has that same error.  I've adjusted the places google said to adjust;  It's like there's some other place I also need to adjust, but I don't know which setting, or that it's improperly holding onto the old info somewhere.

I tried changing the boot java runtime for IDEA to use the jdk 16 instead of the default Jetbrains one, and now it doesn't load at all.  So I have to revert that somehow, then I still need a fix for the actual issue of getting it to use the jdk16 for actually compiling and running the triplea project.

**Comment by zlefin** [comment]
Adding various responses as I find out more;  for the prior error, I had to go to View -> Tool Windows -> Gradle
and from there find the button "Build Tool Settings" and adjust something in there so that it used a more recent java version for gradle building.

That got me past that part, now the problem I'm facing is that when I try to actually run the game via IDEA, it generates this error:
"C:\Program Files\Java\jdk-16.0.1\bin\java.exe" "-javaagent:C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.1.1\lib\idea_rt.jar=53882:C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.1.1\bin" -Dfile.encoding=UTF-8 org.triplea.http.LobbyServer
Error: Could not find or load main class org.triplea.http.LobbyServer
Caused by: java.lang.ClassNotFoundException: org.triplea.http.LobbyServer

Process finished with exit code 1

which makes it seem like the IDEA settings are simply incorrect somewhere so that it's not actually finding the classes because it's not looking in the right place.

**Reply by DanVanAtta** [reply]
The guide to getting setup should be: https://github.com/triplea-game/triplea/blob/master/docs/development/README.md

If the documents from that location are not up to date, we need to get them up to date. Others will have the same problem. On the other hand, the guide does assume some level of base knowledge, so we should not document necessarily everything (in other words, it is not a "teach you Java guide").

Java11 is the version to use. The launcher to deploy the lobby is checked in, so you should be able to see that from the "Run > Edit Configurations" menu. Ensure that you've updated to the very latest code. If the above launcher is one you created, it'll be out of date with the latest.

**Reply by zlefin** [reply]
It seems like everything is working fine now that I'm using the most uptodate version cloned from git.  I think all the issues were due to using the old 22294 stable build, and that it might've had some stuff that since got fixed/correct in the default settings for IDEs or something.  At any rate; it works fine now, so good enough.
